File: 20242Q/poo/taller/taller03/ej1/src/SubwayCard.java
public abstract class SubwayCard {

    private SubwayCentral subwayCentral;

    SubwayCard(SubwayCentral sc) {
        this.subwayCentral = sc;
    }

    protected abstract boolean canRide();

    protected SubwayCentral getSubwayCentral() {
        return subwayCentral;
    }

    public abstract void ride();
}

File: 20242Q/poo/laboratorio/tp08/ej6/subway_card.rb
require_relative 'subway_central'

class SubwayCard
  @subway_central # !! declaro que debe tener una central

  def initialize
    raise 'No se puede instanciar esto'
  end

  private def ride?
    raise 'Implementar'
  end

  def ride
    raise 'Implementar'
  end

  def to_s
    "Subway Card [Central: #{@subway_central}]"
  end
end